### PR TITLE
select editor: add columnrowdata option

### PR DESCRIPTION
**Pull request by matallo**
before moving on with the tests, I fixed the issue of filling the select editor with the results from `columnRowData` passing it as an option (instead of going full new component)

if the item selected isn't in the `columnRowData` results it's added to the collection

CR @xavijam (I'd add the tests for this along https://github.com/CartoDB/cartodb/pull/10713 PR)

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-view.js [hunk 1]
```diff
@@ -35,14 +35,36 @@ Backbone.Form.editors.Select = Backbone.Form.editors.Base.extend({
...
+    this.setValue(this.value);
+
+    return this;
+  },
+
+  _setupCollection: function () {
+    this.collection = new CustomListCollection(this.options.options);
+
+    if (this.options.editorAttrs && this.options.editorAttrs.columnRowData) {
+      var columnRowData = this.options.editorAttrs.columnRowData;
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-view.js]: No references to columnRowData please, let's use a genertic name, like collection or collectionOptions.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-view.js [hunk 2]
```diff
@@ -35,14 +35,36 @@ Backbone.Form.editors.Select = Backbone.Form.editors.Base.extend({
...
+      columnRowData.fetch();
+      columnRowData.bind('columnsFetched', function (rows) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-view.js]: Bind should be before the fetch.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-view.js [hunk 3]
```diff
@@ -81,6 +103,9 @@ Backbone.Form.editors.Select = Backbone.Form.editors.Base.extend({
 
   _initBinds: function () {
     this.collection.bind('change:selected', this._onItemSelected, this);
+    this.collection.bind('reset', function () {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/select/select-view.js]: this.collection.bind('reset', this.render, this);

File: lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-model.js [hunk 4]
```diff
@@ -73,22 +61,12 @@ module.exports = Backbone.Model.extend({
...
-              }
-            };
-
-            self.trigger('changeSchema');
-          });
+
+          self.schema[columnName].editorAttrs = {
+            showSearch: true,
+            allowFreeTextInput: true,
+            columnRowData: columnRowData
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/edit-feature-content-views/edit-feature-attributes-form-model.js]: Remember, better to use a generic name, not that one 💋 .

**Comment by matallo** [comment]
- applied the suggested changes (still some code outside the editor to make the Text => Select change)
- created a new 'Suggest' editor which I will further work on
- made the tests pass

when we are 💚 I'll merge and keep working in the other branch: creating the tests, and extending the 'Suggest' editor (so we don't have to make a render of the whole form)

**Pull request merged**